### PR TITLE
Factor out a handler for async operations to decouple code.

### PR DESCRIPTION
**Pull request by baldersheim**
@geirst @vekterli PR

### CODE REVIEW

File: storage/src/vespa/storage/persistence/testandsethelper.cpp [hunk 1]
```diff
@@ -34,25 +35,20 @@ void TestAndSetHelper::parseDocumentSelection(const document::DocumentTypeRepo &
 }
 
 spi::GetResult TestAndSetHelper::retrieveDocument(const document::FieldSet & fieldSet, spi::Context & context) {
-    return _thread._spi.get(
-        _thread.getBucket(_docId, _cmd.getBucket()),
-        fieldSet,
-        _cmd.getDocumentId(),
-        context);
+    return _spi.get(_env.getBucket(_docId, _cmd.getBucket()), fieldSet,_cmd.getDocumentId(),context);
```
Comment (vekterli) [inline storage/src/vespa/storage/persistence/testandsethelper.cpp:38]: Nit: missing spacing after commas

**Review by vekterli — APPROVED** [review]
:+1:

File: storage/src/vespa/storage/persistence/testandsethelper.cpp (hunk 1 above)
Comment (baldersheim) [reply storage/src/vespa/storage/persistence/testandsethelper.cpp:38]: Fixed in upcomming PR

File: storage/src/vespa/storage/persistence/asynchandler.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+namespace vespalib { class ISequencedTaskExecutor; }
+namespace storage {
+
+namespace spi {
+    struct PersistenceProvider;
+    class Context;
+}
+struct PersistenceUtil;
+
+class AsyncHandler : public Types {
```
Comment (geirst) [inline storage/src/vespa/storage/persistence/asynchandler.h:16]: Consider adding class comment.

**Review by geirst — COMMENTED** [review]
👍

File: storage/src/vespa/storage/persistence/asynchandler.h (hunk 2 above)
Comment (baldersheim) [reply storage/src/vespa/storage/persistence/asynchandler.h:16]: Fixed in #14929

**Pull request merged**